qualtran/bloqs/generalized_qsp.py

```diff
@@ -115,9 +197,9 @@ def qsp_phase_factors(
     for d in reversed(range(n)):
         assert S.shape == (2, d + 1)
 
-        a, b = np.around(S[:, d], decimals=10)
+        a, b = S[:, d]
         theta[d] = np.arctan2(np.abs(b), np.abs(a))
-        phi[d] = (np.angle(a) - np.angle(b)) % (2 * np.pi)
+        phi[d] = np.angle(a) - np.angle(b)
```

Comment:
Can you add a comment with a TODO that highlights the numerical instability for complex polynomials and what causes the numerical instability? (or link to a github issue that describes the problem)

Reply:
added a comment before `phi[d]`